android/java/org/chromium/chrome/browser/shields/BraveShieldsScreenshotUtil.java

```diff
@@ -0,0 +1,128 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.shields;
+
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Bitmap.CompressFormat;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.chromium.base.ThreadUtils;
+import org.chromium.base.task.AsyncTask;
+import org.chromium.chrome.browser.feedback.ScreenshotTask;
+
+import java.io.ByteArrayOutputStream;
+
+public class BraveShieldsScreenshotUtil {
+    private static final String TAG = "ShieldsScreenshot";
+    private final BraveScreenshotRunnable mBraveScreenshotRunnable;
+
+    private static class PngConvertorTask extends AsyncTask<byte[]> {
```

Comment:
Asynctask is deprecated with api 30. we should use executors here.

Reply:
Good explanation on why we shouldn't use asynctask. : https://proandroiddev.com/a-quick-story-about-async-callbacks-memory-leaks-weakreferences-and-misconceptions-78003b3d6b26

Reply:
done